core/src/main/java/org/apache/gravitino/catalog/CatalogManager.java

```diff
@@ -677,11 +678,70 @@ public boolean dropCatalog(NameIdentifier ident, boolean force)
     } catch (NoSuchMetalakeException | NoSuchCatalogException ignored) {
       return false;
 
-    } catch (IOException e) {
+    } catch (GravitinoRuntimeException e) {
+      throw e;
+
+    } catch (Exception e) {
       throw new RuntimeException(e);
     }
   }
 
+  /**
+   * Check if the given list of schema entities contains any currently existing user-created
+   * schemas.
+   *
+   * <p>This method determines if there are valid user-created schemas by comparing the provided
+   * schema entities with the actual schemas currently existing in the external catalog. It
```

Comment:
Better changing to "external data source"

Reply:
updated